src/configValidator/index.js

```diff
@@ -1,21 +1,25 @@
+import {groupBy} from 'lodash'
 import checkConfig from '../checkConfig'
 import uncoveredPaths from '../uncoveredPaths'
-import messageFormatter from '../messageFormatter'
+import formatMessagesAsString from '../formatMessagesAsString'
 
 export default configValidator
 
 function configValidator(configName, config, validators) {
-  const uncoveredFields = getWarningsForUncoveredFields(config, validators)
-  const results = checkConfig(config, validators)
-  const totalResults = uncoveredFields.concat(results)
-  if (totalResults.length) {
-    const {error, warning} = messageFormatter(totalResults)
-    log(configName, error)
-    log(configName, warning)
+  const uncoveredFieldsMessages = getWarningMessagesForUncoveredFields(config, validators)
+  const validationMessages = checkConfig(config, validators)
+  const messages = [...uncoveredFieldsMessages, ...validationMessages]
+  const {warning: warnings = [], error: errors = []} = groupBy(messages, 'type')
```

Comment:
Can we use `warnings` for for destructing refs, it would be better to have consistency in convention. Feel free to defer it.